resources/developer-mode.mjs

```diff
@@ -96,15 +161,28 @@ function fixupURL() {
         url.searchParams.delete("suites");
         url.searchParams.delete("suite");
         url.search = decodeURIComponent(url.search);
-
-        // Only push state if changed
-        if (url.href !== window.location.href)
-            window.history.pushState({}, "", url);
     } else {
-        const url = new URL(window.location.href);
         url.searchParams.delete("suite");
         url.searchParams.set("suites", selectedSuites.join(","));
-        url.search = decodeURIComponent(url.search);
-        window.history.pushState({}, "", url);
     }
+
+    if (params.measurementMethod === "raf")
+        url.searchParams.set("measurementMethod", "raf");
+    else
+        url.searchParams.delete("measurementMethod");
+
+    if (params.iterationCount !== defaultParams.iterationCount)
+        url.searchParams.set("iterationCount", params.iterationCount);
+    else
+        url.searchParams.delete("iterationCount");
+
+    if (params.useWarmupSuite !== defaultParams.useWarmupSuite)
+        url.searchParams.set("useWarmupSuite", params.useWarmupSuite);
+    else
+        url.searchParams.delete("useWarmupSuite");
```

Comment:
For a separate PR maybe: This encoding should probably go on the `Params` object directly.

Something like: `url.search = params.encodeSearchParams().toString()`